File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main (void) {
    string text = get_string("Text:\n");

    int letters, words, sentences, length;

    letters = 0;
    words = 1;
    sentences = 0;

    length = strlen(text);

    for (int i = 0; i < length; i++) {
        if (isalpha(text[i])) {
            letters++;
        }
    }

    for (int j = 0; j < length; j++) {

        if (j != 0 && text[j] == ' ') {
            words ++;
        }
    }

     for (int k = 0; k < length; k++) {

        if (text[k] == '!' || text[k] == '.'|| text[k] == '?') {
            sentences ++;
        }
    }

    float L = ((float)letters/(float) words) * 100;
    float S = ((float)sentences/(float) words) * 100;

   int index = round((0.0588 * L) - (0.296 * S) - 15.8);

   if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 0)
    {
        printf("Before Grade 1\n");
    }
    else {
        printf("Grade %d\n", index);
    }
}

File: pset6/readability/readability.py
# Collecting user-input
text = input("Text: ")

# Collecting values needed to calculate Coleman-Liau index
letters = 0
words = 1
sentences = 0

# Getting length of user input to use for loops
length = len(text)

# Loop checking for alphabet values
for i in range(length):
    if (text[i].isalpha() == True):
        letters = letters + 1

# Loop checking for number of words in user input
for j in range(length):
    if j != 0 and text[j] == ' ':
        words = words + 1;

# Loop for checking number of sentences
for k in range(length):
    if text[k] == '!' or text[k] == '.'or text[k] == '?':
        sentences = sentences + 1

# Index calculations
L = (float(letters)/float(words)) * 100
S = (float(sentences)/float(words)) * 100
index = round((0.0588 * L) - (0.296 * S) - 15.8);

# Conditions to output reading grade level of user input
if index >= 16:
    print("Grade 16+")

elif index <= 0:
    print("Before Grade 1")

else:
    print("Grade " + str(index))